rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -2281,11 +2281,23 @@ extern "C"
 		return RMW_RET_ERROR;
 	}
 	//Get participant pointer from node
-        if(node->implementation_identifier != eprosima_fastrtps_identifier)
-        {
-            RMW_SET_ERROR_MSG("node handle not from this implementation");
-            return RMW_RET_ERROR;
-        }
+	if(node->implementation_identifier != eprosima_fastrtps_identifier)
+	{
+	    RMW_SET_ERROR_MSG("node handle not from this implementation");
+	    return RMW_RET_ERROR;
+	}
+	if (topic_names_and_types->topic_count) {
```

Comment:
This could probably be addressed in rcl. Currently the docs don't mentioned "zero initialized" and only mention that the structure pointer should point to allocated space:

https://github.com/ros2/rcl/blob/master/rcl/include/rcl/graph.h#L35-L58

We could push this check up to rcl and/or ensure it is setup properly before passing to rmw.

The whole point of the "zero initialized" requirement I have for structure pointers throughout rcl is to catch exactly this case, i.e. create_x fails and so the memory is still uninitialized and the user passes the allocated, but uninitialized, structure to destroy which assumes it is not pointing to bogus data.